common/meta/raft-store/src/state_machine/sm_meta_api_impl.rs

```diff
@@ -352,6 +359,61 @@ impl MetaApi for StateMachine {
         Ok(UpsertTableOptionReply {})
     }
 
+    async fn create_share(&self, req: CreateShareReq) -> Result<CreateShareReply, MetaError> {
+        let tenant = &req.tenant;
+        let share_name = &req.share_name;
+        let if_not_exists = req.if_not_exists;
+
+        tracing::info!("create share: {:}: {:?}", &tenant, &share_name);
+
+        let cr = Cmd::CreateShare {
+            tenant: tenant.clone(),
+            share_name: share_name.clone(),
+        };
```

Comment:
For a historical reason, `Cmd` does not utilize `XXXReq`. But instead in our codebase it manually converts `XXXReq` to a corresponding `Cmd`.

It would be convenient to let `Cmd` have a variant that includes an `XXXReq` in it. E.g.: 
```rust
enum Cmd {
    CreateShare(CreateShareReq),
}
```

This would save a lot of trouble. :D

Reply:
is it time to do it now? can I do the refactor by the way?

Reply:
Next time will be a good time:)